client/reducers/games.js

```diff
@@ -46,12 +46,15 @@ function games(state = {
                 currentGame: action.currentGame
             });
 
-            var currentState = state.currentGame;
+            var currentState = retState.currentGame;
+            if(!currentState) {
+                return retState;
+            }
 
-            if(currentState && currentState.players[action.username] && currentState.players[action.username].left) {
+            if(!currentState || !currentState.players[action.username] || currentState.players[action.username].left) {
```

Comment:
This change makes spectating impossible. When joining as a spectator, it hits this line and immediately deletes the game state since spectators are not in the players list.